diesel/src/type_impls/date_and_time.rs

```diff
@@ -35,16 +35,16 @@ mod chrono {
         diesel(sql_type = crate::sql_types::Timestamptz)
     )]
     #[cfg_attr(feature = "mysql_backend", diesel(sql_type = crate::sql_types::Datetime))]
-    // TODO: Add TimestamptzSqlite stuff here??
+    #[cfg_attr(feature = "sqlite", diesel(sql_type = crate::sql_types::TimestamptzSqlite))]
     struct NaiveDateTimeProxy(NaiveDateTime);
 
-    #[derive(FromSqlRow)]
+    #[derive(AsExpression, FromSqlRow)]
     #[diesel(foreign_derive)]
+    #[diesel(sql_type = Timestamp)]
```

Comment:
We should not allow converting a chrono value with timezone to a `Timestamp` sql value without timestamp.  (Just remove that line)

Reply:
Ah yeah; unfortunately when I remove this line I get the following error message when running `bin/test`
```
+ cd diesel_tests
+ cargo test --features mysql --no-default-features -- --test-threads 1
   Compiling diesel_derives v2.0.0 (/Users/stevenchu/personal_projects/rust/diesel/diesel_derives)
   Compiling diesel v2.0.0 (/Users/stevenchu/personal_projects/rust/diesel/diesel)
error: At least one `sql_type` is needed for deriving `AsExpression` on a structure.
  --> diesel/src/type_impls/date_and_time.rs:41:14
   |
41 |     #[derive(AsExpression, FromSqlRow)]
   |              ^^^^^^^^^^^^
   |
   = note: this error originates in the derive macro `AsExpression` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error

error: could not compile `diesel`

To learn more, run the command again with --verbose.
```

Unless there's an easy fix here, I think I probably need to step back some and understand better the Rust ecosystem -- I've been mostly copy pasting and fixing things iteratively after `cargo check` runs; but I think there's a lot going on here that maybe I need to get more in depth on 😅 . At some point may just reach out in gitter to get some more direct help too

Reply:
You should be able to fix that by putting the `#[derive(AsExpression)]` behind the following cfg feature gate:
`#[cfg_attr(any(feature = "postgres_backend", feature = "sqlite"), derive(AsExpression))]`
This should enforce that there is only a single derive even if both features are enabled + there is always a corresponding `#[diesel(sql_type)]` if the derive is added.

Reply:
Ahhh perfect thank you so much :) everything compiles now; going to work on a few more tests and then push up the next commit and mark it ready for review!!